### PR TITLE
Unify asset identification to Uuid structs

### PR DESCRIPTION
**Pull request by Frizi**
Depends on https://github.com/kabergstrom/amethyst/pull/2

### CODE REVIEW

File: src/asset_hub.rs [hunk 1]
```diff
@@ -126,15 +100,12 @@ impl AssetHub {
         Ok(cursor)
     }
 
-    pub fn get_metadata<'a, K, V: DBTransaction<'a, T>, T: lmdb::Transaction + 'a>(
+    pub fn get_metadata<'a, V: DBTransaction<'a, T>, T: lmdb::Transaction + 'a>(
```
Comment (LucioFranco) [inline src/asset_hub.rs]: I know you didn't add this but I think the compiler can infer this lifetime now in 2018?

File: src/file_asset_source.rs [hunk 2]
```diff
@@ -159,8 +160,8 @@ where
 
 fn calc_import_hash(
     scratch_buf: &mut Vec<u8>,
-    importer_options: &Box<SerdeObj>,
-    importer_state: &Box<SerdeObj>,
+    importer_options: &dyn SerdeObj,
```
Comment (LucioFranco) [inline src/file_asset_source.rs]: 👍

File: src/asset_hub.rs (hunk 1 above)
Comment (Frizi) [reply src/asset_hub.rs]: Maybe it could here, I haven't tried. But anyway, this should probably not be done as a part of this PR.

Comment (LucioFranco) [reply src/asset_hub.rs]: 👍 Yeah, just kinda stuck out to me I guess I'm too indoctrinated in 2018 edition.

**Comment by kabergstrom** [comment]
Thanks a lot for this! The 2018 cleanup, clippy fixes after crossbeam_channel upgrade and more are all looking really good.

I don't have anything to add in this PR, let's merge it!

File: importer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+name = "importer"
+version = "0.1.0"
+edition = "2018"
+
+[dependencies]
+uuid = { version = "0.7", features = ["v4", "serde"] }
+downcast = "0.9.2"
+serde = "1.0"
+serde_dyn = "0.2.2"
+amethyst = { path = "../../amethyst"}
```
Comment (kabergstrom) [inline importer/Cargo.toml]: Perhaps this should be a github repo?

**Pull request merged**